### PR TITLE
fix closest_point_on_segment_xy

### LINKED ISSUE #837: closest_point_on_segment_xy returns a 3D point (Z corrdinate !=0)
**Describe the bug**

If the `closest_point_on_segment_xy` is one of the endpoints, it is returned with z coordinate different than 0.
This is because the [logic returns the endpoint as-is](https://github.com/compas-dev/compas/blob/961b15e7560d2b450e6caaa77ad5676af4103beb/src/compas/geometry/_core/distance.py#L766), without zeroing the Z coordinate.

**To Reproduce**

```py
from compas.geometry import Point, Line, closest_point_on_segment_xy


def test_closest_point_segment_xy():
    point = Point(0, 0, -10)
    line = Line([1, 1, -15], [10, 10, 20])
    ponl= closest_point_on_segment_xy(point, line)
    assert ponl == Point(1, 1, 0)
```

it returns

```
=============================== FAILURES ================================
_________________ test_closest_point_segment_xy ________________________

    def test_closest_point_segment_xy():
        point = Point(0, 0, -10)
        line = Line([1, 1, -15], [10, 10, 20])
        ponl = closest_point_on_segment_xy(point, line)
>       assert ponl == Point(1, 1, 0)
E       assert Point(1.000, 1.000, -15.000) == Point(1.000, 1.000, 0.000)
E         Use -v to get the full diff
```

### PR DESCRIPTION
**Pull request by sanzoghenzo**
fixes #837

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/compas/geometry/test_distance.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from compas.geometry import Point
+from compas.geometry import Line
+from compas.geometry import closest_point_on_segment_xy
+
+
+def test_closest_point_segment_xy():
```
Comment (tomvanmele) [inline tests/compas/geometry/test_distance.py]: would you mind including non-boundary cases as well?

**Review by tomvanmele — COMMENTED** [review]
cool, thanks for contributing!
just a small request about non-boundary cases....

File: tests/compas/geometry/test_distance.py (hunk 1 above)
Comment (sanzoghenzo) [reply tests/compas/geometry/test_distance.py]: There you have it, I parametrized the test and added a non boundary case.

**Review by tomvanmele — APPROVED** [review]
